mne_bids/write.py

```diff
@@ -1352,3 +1353,162 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
+    overwrite : bool
+        If ``False``, only update the information of the channels passed via
+        ``channels``, and leave the rest untouched. If ``True``, update the
+        information of **all** channels: mark the channels passed via
+        ``channels`` as bad, and all remaining channels as good, also
+        discarding their descriptions.
+    verbose : bool
+        The verbosity level.
+
+    Raises
+    ------
+    ValueError
+        If the number of channels and descriptions does not match.
+
+    ValueError
+        If the specified channels cannot be found in the dataset.
+
+    ValueError
+        If the specified ``kind`` cannot be found in the dataset.
+
+    ValueError
+        If ``overwite=False``, but no channel names were passed.
+
+    RuntimeError
+        If multiple recording kinds are present in the dataset, but
+        ``kind=None``.
+
+    RuntimeError
+        If more than one data files exist for the specified recording.
+
+    Examples
+    --------
+    Mark a single channel as bad.
+
+    >>> mark_bad_channels('MEG 0112', bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark multiple channels as bad.
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and add a description as to why.
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> descriptions = ['Only produced noise', 'Continuously flat']
+    >>> mark_bad_channels(bads, descriptions, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and reset all others existing in the dataset to
+    "good".
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, overwrite=True, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark all channels as good.
+
+    >>> mark_bad_channels([], bids_basename=bids_basename, bids_root=bids_root)
+
+    """
+    if not channels and not overwrite:
+        raise ValueError('You did not pass a channel name, but set '
+                         'overwrite=False. If you wish to mark all channels '
+                         'as good, please pass overwrite=True')
+
+    if descriptions and not channels:
+        raise ValueError('You passed descriptions, but no channels.')
+
+    if not channels:
+        descriptions = []
+    if isinstance(channels, str):
+        channels = [channels]
+    if isinstance(descriptions, str):
+        descriptions = [descriptions]
+    elif descriptions is None:
+        descriptions = ['n/a'] * len(channels)
+    if len(channels) != len(descriptions):
+        raise ValueError('Number of channels and descriptions must match.')
+
+    # convert to BIDSPath
+    if isinstance(bids_basename, str):
+        params = parse_bids_filename(bids_basename)
+        bids_basename = BIDSPath(subject=params.get('sub'),
```

Comment:
```suggestion
        bids_basename = BIDSPath(**params)
```
?

Reply:
Unfortunately this won't work as the keys in `params` are diffeent than what `BIDSPath` expects (e.g. `sub` vs `subject` etc)